libsql-ffi/bundled/src/sqlite3.h

```diff
@@ -13639,11 +13640,26 @@ struct libsql_wal {
   unsigned char lockError;            /* True if a locking error has occurred */
   WalIndexHdr *pSnapshot;             /* Start transaction here if not NULL */
   sqlite3 *db;
-  libsql_wal_methods *pMethods;       /* Virtual methods for interacting with WAL */
-  void *pMethodsData;                 /* Optional context for private use of libsql_wal_methods */
+} sqlite3_wal;
+
+struct libsql_wal {
+    libsql_wal_methods methods; /* virtual wal methods */
+    wal_impl* pData; /* methods receiver */
 };
 
-typedef struct libsql_wal libsql_wal;
+typedef struct RefCountedWalManager {
+    int n;
+    libsql_wal_manager ref;
+    int is_static;
+} RefCountedWalManager;
+
+int make_ref_counted_wal_manager(libsql_wal_manager wal_manager, RefCountedWalManager **out);
+void destroy_wal_manager(RefCountedWalManager *p);
+RefCountedWalManager* clone_wal_manager(RefCountedWalManager *p);
+
+void make_sqlite3_wal_manager_rc(RefCountedWalManager **out);
```

Comment:
nitpick: out parameters make sense if you use the return value for error codes. Otherwise just rewrite it to `RefCountedWalManger *make_sqlite3_wal_manager_rc()`, it will be much clearer to follow and more standard